SNIClient.py

```diff
@@ -1131,6 +1129,126 @@ async def main():
     if input_task:
         input_task.cancel()
 
+def get_alttp_settings(romfile: str):
+    lastSettings = Utils.get_adjuster_settings(GAME_ALTTP)
+    adjusted = False
+    adjustedromfile = ''
+    if lastSettings:
+        choice = 'no'
+        if not hasattr(lastSettings, 'auto_apply') or 'ask' in lastSettings.auto_apply:
+
+            whitelist = {"music", "menuspeed", "heartbeep", "heartcolor", "ow_palettes", "quickswap",
+                        "uw_palettes", "sprite", "sword_palettes", "shield_palettes", "hud_palettes",
+                        "reduceflashing", "deathlink"}
+            printed_options = {name: value for name, value in vars(lastSettings).items() if name in whitelist}
+            if hasattr(lastSettings, "sprite_pool"):
+                sprite_pool = {}
+                for sprite in getattr(lastSettings, "sprite_pool"):
+                    if sprite in sprite_pool:
+                        sprite_pool[sprite] += 1
+                    else:
+                        sprite_pool[sprite] = 1
+                    if sprite_pool:
+                        printed_options["sprite_pool"] = sprite_pool
+            import pprint
+
+            if gui_enabled:
+            
+                from tkinter import Tk, PhotoImage, Label, LabelFrame, Frame, Button
+                applyPromptWindow = Tk()
+                applyPromptWindow.resizable(False, False)
+                applyPromptWindow.protocol('WM_DELETE_WINDOW',lambda: onButtonClick())
```

Comment:
without testing admittedly, but I have a feeling `applyPromptWindow.protocol('WM_DELETE_WINDOW', onButtonClick)` should work here

Reply:
Been a few weeks but I do recall not using lambda at first and it did not work as intended, don't remember if the window just closed immediately or if I got a function not defined, but I had to use lambda